text/0038-output-types-for-tokenization-and-sc/0038-output-types-for-tokenization-and-sc.md

```diff
@@ -0,0 +1,3416 @@
+validation.
+
+The different output types and optional output feature blocks add extra constraints to transaction validation rules,
+but since these are specific to the given outputs and features, they are discussed for each
+[output types](#output-design) and
+[feature blocks types](#optional-output-features).
+
+#### Syntactic Validation
+
+This validation can commence as soon as the transaction data has been received in its entirety. It validates the
+structure of the transaction. If the transaction does not pass this stage, it must not be broadcast further and can be
+discarded right away.
+
+The following criteria defines whether the transaction passes the syntactic validation:
+* `Transaction Essence Type` value must be 0, denoting an `Transaction Essence`.
+* Inputs:
+    * `Inputs Count` must be 0 < x ≤ `Max Inputs Count`.
+    * At least one input must be specified.
+    * `Input Type` value must be 0, denoting an `UTXO Input`.
+    * `UTXO Input`:
+        * `Transaction Output Index` must be 0 ≤ x < `Max Outputs Count`.
+        * Every combination of `Transaction ID` + `Transaction Output Index` must be unique in the list of inputs.
+* Outputs:
+    * `Outputs Count` must be 0 < x ≤ `Max Outputs Count`.
+    * At least one output must be specified.
+    * `Output Type` must denote a `SimpleOutput`, `ExtendedOutput`, `AliasOutput`, `FoundryOutput` or `NFTOutput`.
+    * Output must fulfill the [dust protection requirements.](https://github.com/iotaledger/protocol-rfcs/pull/39)
+    * Output is syntactically valid based on its type.
+    * Accumulated output balance must not exceed the total supply of tokens `2'779'530'283'277'761`.
+* `Payload Length` must be 0 (to indicate that there's no payload) or be valid for the specified payload type.
+* `Payload Type` must be one of the supported payload types if `Payload Length` is not 0.
+* `Unlock Blocks Count` must match the amount of inputs. Must be 0 < x ≤ `Max Inputs Count`.
+* `Unlock Block Type` must either be 0, 1, 2 or 3, denoting a `Signature Unlock Block`, a `Reference Unlock block`, an
+  `Alias Unlock Block` or an `NFT Unlock Block`.
+* `Signature Unlock Blocks` must define a `Ed25519 Signature`.
+* A `Signature Unlock Block` unlocking multiple inputs must only appear once (be unique) and be positioned at the same
+  index of the first input it unlocks. All other inputs unlocked by the same `Signature Unlock Block` must have a
+  companion `Reference Unlock Block` at the same index as the corresponding input which points to the origin
+  `Signature Unlock Block`.
+* `Reference Unlock Blocks` must specify a previous `Unlock Block` which is not of type `Reference Unlock Block`. The
+  referenced index must therefore be < the index of the `Reference Unlock Block`.
+* `Alias Unlock Blocks` must specify a previous `Unlock Block` which unlocks the alias the input is locked to. The
```

Comment:
For `Alias Unlock Blocks` and `NFT Unlock Blocks`, shouldn't this mention that they should be ordered ? I know it's mentioned later but this is definitely syntactic validation.

Reply:
By ordering, do you mean the fact that they can only point to a smaller index? That constraint is defined here:
> must specify a previous `Unlock Block` ...

Reply:
Later in the RFC there is `Additionally, the Alias Unlock Blocks must also be ordered to prevent circular dependencies:` and same for NFT. I am not sure if that means they should ALL be ordered or not, if yes then this rule is not enough as you can have for example 4 -> 3 and 5 -> 2, they point to lower indexes but they are not ordered.

Reply:
So it is my bad wording at the end of the day :smile_cat: 

By ordering I mean here implicit ordering due to the "must point to a lower index" criteria. You don't need to order the unlock blocks per se, they refer to inputs so you actually order inputs. By requiring pointing to a lower index, we enforce ordering of dependent unlock blocks only that may cause a deadlock. We have had the same conceptual rule btw for the refernce unlock blocks, they can only point to previous unlocks also.

Reply:
Got it thanks, just wanted to be sure 👍